plugin/ochttp/trace_test.go

```diff
@@ -79,6 +79,26 @@ func (t testPropagator) SpanContextToRequest(sc trace.SpanContext, req *http.Req
 	req.Header.Set("trace", hex.EncodeToString(buf.Bytes()))
 }
 
+func TestTransport_RoundTrip_Race(t *testing.T) {
+	// This tests that we don't modify the request in accordance with the
+	// specification for http.RoundTripper.
+	// We attempt to trigger a race by reading the request from a separate
+	// goroutine. If the request is modified by Transport, this should trigger
+	// the race detector.
+
+	transport := &testTransport{ch: make(chan *http.Request, 1)}
+	rt := &Transport{
+		Propagation: &testPropagator{},
+		Base:        transport,
+	}
+	req, _ := http.NewRequest("GET", "http://foo.com", nil)
+	go func() {
+		fmt.Println(*req)
```

Comment:
Sorry am late to the party but I don't think that our tests should be spamming stdout. If we'd like to do that, perhaps use t.Logf.

However, could we actually test that the headers are unmodified after the ochttp.Transport roundtrip?

Reply:
This doesn't actually go to stdout when running `go test` so I don't really see the harm. 

The advantage of this is that we test whether any part of the request (not just the headers) was modified. But in reality you're right that the headers are what actually was being modified.

Reply:
>This doesn't actually go to stdout when running go test so I don't really see the harm.

The harm is spamming stdout. Currently you don't see it but that's because that goroutine is racy(i.e. no synchronization and it isn't guaranteed to run by the time the test exits) and did the race detector trip out at runtime without the fix?

If it were https://play.golang.org/p/Ur72IrWCTQY aka 
```go
package main

import (
	"fmt"
	"testing"
)

func TestFoo(t *testing.T) {
	done := make(chan bool)
	go func() {
		fmt.Println("foo")
		done <- true
	}()
	<-done
}
```
You'd def see the output. Please remove that fmt.Println and we can do
```go
_ = *req
```
if we really would like to keep it :)

Also perhaps for real race triggering at runtime ensure that that that goroutine is actually synchronized and runs concurrently with the code outside it.

Reply:
It only gets printed by "go test -v". Try it.

The code as-is repro'ed the race before I fixed it.

Reply:
We don't need to print this. Most users always use -v when running tests including our CI. We shouldn't pollute the output with arbitrary output.

Reply:
I don't really have strong feelings about this but this PR is merged. The right way to proceed is to open a new issue or a PR for this. Commenting on merged PRs is very difficult to track, I would prefer if we avoid this in future.